packages/workflow/src/worker-interface.ts

```diff
@@ -188,15 +188,20 @@ export function concludeActivation(): coresdk.workflow_completion.IWorkflowActiv
  * @returns number of unblocked conditions.
  */
 export function tryUnblockConditions(): number {
+  const activator = getActivator();
+
+  // If a Workflow Task error was captured, bubble it up now
+  if (activator.workflowTaskError) throw activator.workflowTaskError;
```

Comment:
This won't cover exceptions that are thrown in the condition functions AFAIU.

Reply:
Condition functions are sync. If a condition throws synchronously, the error will bubble up out of `tryUnblockConditions()`, just the same as if they had been thrown from that line.

Microtasks that may get unblocked by the resolving the condition will not execute synchronously, so throwing from there would therefore belong to whatever is waiting on the corresponding Promise, or will result in an unhandled promise rejection, which would be correct in that case.

Reply:
👍